dashboard/src/components/AppView/SecretsTable/SecretsTable.tsx

```diff
@@ -0,0 +1,74 @@
+import * as _ from "lodash";
+import * as React from "react";
+
+import { ErrorSelector } from "../../../components/ErrorAlert";
+import { IKubeItem, ISecret } from "../../../shared/types";
+import SecretItem from "./SecretItem";
+
+interface IServiceTableProps {
+  namespace: string;
+  secretNames: string[];
+  secrets: { [s: string]: IKubeItem };
```

Comment:
perhaps this type could be `IKubeItem<ISecret>` so we can keep the type of the k8s resource?

Reply:
Sorry, my comment didn't go into enough detail here, I meant that we should make the IKubeItem type support a generic type for its `item` key. To do so, requires a change in the type definition. See my comment below.